src/axom/slic/tests/slic_macros.cpp

```diff
@@ -122,6 +122,21 @@ TEST(slic_macros, test_error_macros)
   SLIC_ERROR_IF(false, "this message should not be logged!");
   EXPECT_TRUE(slic::internal::is_stream_empty());
 
+  // Check selective filtering based on root == false
+  axom::slic::setIsRoot(false);
+  SLIC_ERROR_ROOT_IF(false, "this message should not be logged!");
+  EXPECT_TRUE(slic::internal::is_stream_empty());
+
```

Comment:
Is this missing two tests?

```cpp
  // is root, but conditional is false -> no message
  axom::slic::setIsRoot(true);
  SLIC_ERROR_ROOT_IF(false, "this message should not be logged!");
  EXPECT_TRUE(slic::internal::is_stream_empty());
```

```cpp
  // is not root, and conditional is true -> no message
  axom::slic::setIsRoot(fase);
  SLIC_ERROR_ROOT_IF(true"this message should not be logged!");
  EXPECT_TRUE(slic::internal::is_stream_empty());
```

Reply:
Yea I haven't flushed these out at all at this point. It was my initial does this compile test.  More to come!